File: Servicios/Idioma.vb
Imports Entidades
Imports Datos
Imports System.Threading
Imports System.Globalization
Public Class Idioma

    Public Shared Sub AplicarIdioma(ByVal IdiomaUsuario As String)
        Thread.CurrentThread.CurrentUICulture = New CultureInfo(IdiomaUsuario)
    End Sub

    Public Shared Function DetectarIdioma(ByVal IDUsuario As Integer) As Integer
        Return ServicioAD.DetectarIdioma(IDUsuario)
    End Function

    Public Shared Function ListarIdiomas() As List(Of IdiomaEN)
        Return ServicioAD.ListarIdiomas()
    End Function

End Class ' Idioma

File: Servicios/Idioma.cs
using System.Collections.Generic;
using System.Globalization;
using System.Threading;
using Datos;
using Entidades;

namespace Servicios
{
    public class Idioma
    {
        public static void AplicarIdioma(string IdiomaUsuario)
        {
            Thread.CurrentThread.CurrentUICulture = new CultureInfo(IdiomaUsuario);
        }

        public static int DetectarIdioma(int IDUsuario)
        {
            return ServicioAD.DetectarIdioma(IDUsuario);
        }

        public static List<IdiomaEN> ListarIdiomas()
        {
            return ServicioAD.ListarIdiomas();
        }
    }
} // Idioma